### PR TITLE
Docfix/q rdecomposition

### PR DESCRIPTION
**Pull request by bgoodri**
#### Submisison Checklist

- [* ] Run unit tests: `./runTests.py src/test/unit`
- [* ] Run cpplint: `make cpplint`
- [* ] Declare copyright holder and open-source license: see below

#### Summary

Clarify the documentation of the QR decomposition

#### Intended Effect

Make it so the user can know what the dimensions of Q and R are

#### How to Verify

`make manual`

#### Side Effects

Closes https://github.com/stan-dev/stan/issues/1879

#### Documentation

It is documentation

#### Reviewer Suggestions

anyone

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Ben Goodrich

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by bob-carpenter** [comment]
This pull request is just merges --- it's missing the content.

**Comment by bgoodri** [comment]
OK. Pushed now. I don't know what went wrong before.

File: src/docs/stan-reference/functions.tex [hunk 1]
```diff
@@ -2849,22 +2849,26 @@ \subsubsection{QR Decomposition}
...
 %
 \fitem{matrix}{qr\_Q}{matrix \farg{A}}{
-The orthogonal matrix in the QR decomposition of \farg{A}}
+The orthogonal matrix in the fat QR decomposition of \farg{A}, which
+implies that the resulting matrix is square with the same number of
+rows as \farg{A}}
 %
 \fitem{matrix}{qr\_R}{matrix \farg{A}}{
-  The upper triangular matrix in the QR decomposition of \farg{A}}
+  The upper trapazoidal matrix in the fat QR decomposition of \farg{A},
```
Comment (bob-carpenter) [inline src/docs/stan-reference/functions.tex]: trapazoidal -> trapezoidal

File: src/docs/stan-reference/functions.tex [hunk 2]
```diff
@@ -2849,22 +2849,26 @@ \subsubsection{QR Decomposition}
...
+  which implies that the resulting matrix has the same dimensions as
+  \farg{A}}
 %
 \end{description}
 %
 Multiplying a column of an orthogonal matrix by $-1$ still results in 
 an orthogonal matrix, and you can multiply the corresponding row of 
-the upper triangular matrix by $-1$ without changing the product. Thus,
+the upper trapazoidal matrix by $-1$ without changing the product. Thus,
```
Comment (bob-carpenter) [inline src/docs/stan-reference/functions.tex]: ditto

File: src/docs/stan-reference/functions.tex [hunk 3]
```diff
@@ -2849,22 +2849,26 @@ \subsubsection{QR Decomposition}
...
 Stan adopts the normalization that the diagonal elements of the upper
-triangular matrix are strictly positive and the columns of the 
-orthogonal matrix are rescaled if necessary. The input matrix $A$ need
+trapazoidal matrix are strictly positive and the columns of the
```
Comment (bob-carpenter) [inline src/docs/stan-reference/functions.tex]: find & replace?

**Comment by bob-carpenter** [comment]
Other than needing to fix spelling of "trapezoidal", this is good to go.

**Pull request merged**